pinot-common/src/main/java/org/apache/pinot/common/utils/ServiceStatus.java

```diff
@@ -242,22 +240,19 @@ public synchronized Status getServiceStatus() {
         return _serviceStatus;
       }
       long now = System.currentTimeMillis();
-      int numConsumingSegmentsNotCaughtUp = _getNumConsumingSegmentsNotReachedTheirLatestOffset.get();
+      boolean isConsumingSegmentsCounterProvided = _getNumConsumingSegmentsNotReachedTheirLatestOffset != null;
```

Comment:
Is it possible that the config is turned on, but the num segments not reached latest is zero for now (but can get to a higher value later, say on the next service status call)?

If so, it may be better to check the config here than the state.

Reply:
That can't happen. Once a segment reaches to its latest offset, its name gets added to a helper set called `_caughtUpSegments`. In next service status calls, caught up segments will be skipped.